File: client/src/types/JsonModels/Workouts/WorkoutModel.ts
import type { JsonModels } from "../../backend";

export interface WorkoutModel
{
	id: number;
	workoutTemplateId?: number;
	templateName?: string;
	title: string;
	startedAt: string;
	finishedAt?: string;
	durationSeconds?: number;
	totalVolumeKg?: number;
	notes?: string;
	exerciseCount: number;
	setCount: number;
	groups: JsonModels.Workouts.WorkoutExerciseGroupModel[];
}

File: server/FitMate.Core/JsonModels/Workouts/WorkoutModel.cs
namespace FitMate.Core.JsonModels.Workouts;

public class WorkoutModel
{
    public long Id { get; set; }
    public long? WorkoutTemplateId { get; set; }
    public string? TemplateName { get; set; }
    public string Title { get; set; } = string.Empty;
    public DateTime? StartedAt { get; set; }
    public DateTime? FinishedAt { get; set; }
    public int? DurationSeconds { get; set; }
    public decimal? TotalVolumeKg { get; set; }
    public string? Notes { get; set; }
    public int ExerciseCount { get; set; }
    public int SetCount { get; set; }
    public List<WorkoutExerciseGroupModel> Groups { get; set; } = [];
}
